File: adk/tests/fixtures/integration_profiles/go-integration/project/service.go
package checkout

import "errors"

type CheckoutService struct {
	repository *InventoryRepository
}

func NewCheckoutService(repository *InventoryRepository) *CheckoutService {
	return &CheckoutService{repository: repository}
}

func (s *CheckoutService) Checkout(sku string, quantity int) (string, error) {
	if !s.repository.Reserve(sku, quantity) {
		return "", errors.New("insufficient stock")
	}
	return "reserved:" + sku + ":1", nil
}

File: adk/tests/fixtures/integration_profiles/python-integration/project/app/service.py
from app.repository import InventoryRepository


class CheckoutService:
    def __init__(self, repository: InventoryRepository):
        self._repository = repository

    def checkout(self, sku: str, quantity: int) -> str:
        if not self._repository.reserve(sku, quantity):
            raise ValueError("insufficient stock")
        return f"reserved:{sku}:{quantity}"
